src/Internal/DependencyManagement/DelegateContainer/LegacyContainer.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Class LegacyContainer
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\DependencyManagement\DelegateContainer;
+
+use Psr\Container\ContainerInterface;
+use WC_Payments;
+
+/**
+ * WooPayments Legacy Container Delegate.
+ *
+ * This class is a proxy between `src` and `includes`, allwing
+ * `includes` classes to be used as dependencies.
+ */
+class LegacyContainer implements ContainerInterface {
+	/**
+	 * Finds an entry of the container by its identifier and returns it.
+	 *
+	 * @param string $id Identifier of the entry to look for.
+	 * @return mixed Entry.
+	 */
+	public function get( $id ) {
+		$method = $this->transform_class_to_method( $id );
+		return $this->$method();
```

Comment:
I don't think we would blindly call `$this-> method()` here. 

Instead, use `$this->has($id)` to check before calling it, and then throw an exception `WCPay\Internal\DependencyManagement\ContainerException` in case `has` method return `false`.